### DISCUSSION TITLE
I'm receiving a 'non-zero exit status 137' error when running Nuclei as part of a script

### DISCUSSION THREAD

**Discussion by eitansha** [projectdiscovery/nuclei#4831] [General]
Hi all, 

When I run Nuclei manually, everything goes well. 

However, when I call it indirectly through a script that performs other tasks in between, I receive the following error from Nuclei:
'non-zero exit status 137'. Why does this happen?

The parameters/flags that I use are '-list' and '-o'.

**Comment by dogancanbakir** [comment]
Can you share your script? You're receiving an `exit code 137` error, which could be directly related to your script's memory usage.

**Comment by eitansha** [comment]
Yes, here's the function that triggers nuclei:


```
def run_additional_tools():
    with open('sorted_subdomains.txt', 'r') as file:
        domains = file.read().splitlines()

    for domain in domains:
        domain = domain.replace('https://', '')  # Remove https:// for tool compatibility
        urls_file = f"{domain}_200_urls.txt"
        
        print(f"Running additional tools for domain: {domain}...")
        # Katana
        execute_command(f"katana -list {urls_file} -o {domain}_katana_results.txt")
        # Nuclei
        execute_command(f"nuclei -list {urls_file} -o {domain}_nuclei_results.txt")

```

**Comment by dogancanbakir** [comment]
I have reviewed the code snippet you shared and cannot determine the exact cause of the OOM error you mentioned. However, I noticed that you are also running Katana, which could be a contributing factor to the error. To avoid such errors, ensure that your system has enough memory to support running multiple tools concurrently and storing the necessary data in memory.

**Comment by eitansha** [comment]
Do you think running nuclei as the last tool will resolve this issue?

**Comment by dogancanbakir** [comment]
[This is my only suggestion](https://github.com/projectdiscovery/nuclei/issues/4824#issuecomment-1976149830) since I don't know all the details -script, resource, etc.

**Comment by eitansha** [comment]
I'm running this script on Ubuntu with 2 VCPUs and 2 GB of RAM. Might this not be enough?

**Comment by dogancanbakir** [comment]
Better. It seems like you have tried running nuclei and did not encounter any errors. This suggests that the memory capacity of your system is sufficient for running nuclei. Similarly, if you run a script that includes only one tool, for instance, "nuclei", and it runs without any errors, it means that the memory capacity is also enough for running "nuclei" from your script. The same goes for "katana" as well. Therefore, I would suggest running one tool at a time to see how it performs. If the issue persists, try increasing the system resources and retry.

**Comment by eitansha** [comment]
I'll try what you suggest. I think I'll have some 'sleeps' between the tools running. Thank you!

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
